lib/puppet/network/formats.rb

```diff
@@ -64,6 +64,8 @@ def supported?(klass)
 
 Puppet::Network::FormatHandler.create(:s, :mime => "text/plain", :extension => "txt")
 
+Puppet::Network::FormatHandler.create(:binary, :mime => "application/octet-stream")
+
 # A very low-weight format so it'll never get chosen automatically.
 Puppet::Network::FormatHandler.create(:raw, :mime => "application/x-raw", :weight => 1) do
   def intern_multiple(klass, text)
```

Comment:
I think we should also remove `:raw` format, and change `Puppet::FileServing::Content` to use `:binary` as its supported format, and rename its `from_raw` & `to_raw` methods to `from_binary` & `to_binary`.